functions/src/controllers/campaign.js

```diff
@@ -16,12 +18,63 @@ const Campaign = {
 	 * @param {object} res - response object
 	 * @return  {Object} result
     */
+   
+    async adminGetAll(req, res) {
+        try {
+            const campaigns = [];
+            const data = await db.collection('campaigns').get();
+            const docs = data.docs;
+            for (const doc of docs) {
+                const selectedItem = {
+                    id: doc.id,
+                    campaign: doc.data(),
+                };
+                campaigns.push(selectedItem);
+            }
+            return successNoMessage(res, OK, campaigns);
+        } catch (error) {
+            tryCatchError(res, error);
+        }
+    },
+
+    async adminGetOne(req, res) {
+        try {
+            const document = db.collection('campaigns').doc(req.params.id);
+            if (!document) {
+                return validationError(res, 'Document not found');
+            }
+            const documentData = await document.get();
+            const campaign = documentData.data();
+            return successNoMessage(res, OK, campaign);
+        } catch (error) {
+            tryCatchError(res, error);
+        }
+    },
+
+    async campaignApproved(req, res) {
+        try {
+            const document = db.collection('campaigns').doc(req.params.id);
+            if (!document) validationError(res, errors);
+            req.body.status = approved;
+            req.body.staus = live;
```

Comment:
There is a mix up here i guess

Reply:
It should only be approved until its been sent to pipul tv before we update status to live

Reply:
fix this and then the code is good to merge